presto-orc/src/main/java/com/facebook/presto/orc/reader/DoubleStreamReader.java

```diff
@@ -321,39 +309,32 @@ public void scan()
         endScan(presentStream);
     }
 
-    void addNullResult()
+    private void addResult(double value)
     {
-        if (outputChannel != -1) {
-            if (valueIsNull == null) {
-                valueIsNull = new boolean[values.length];
-            }
-            ensureResultRows();
-            valueIsNull[numResults + numValues] = true;
+        if (outputChannel == -1) {
+            return;
         }
-        numResults++;
-    }
 
-    void addResult(double value)
-    {
         int position = numValues + numResults;
-        if (position >= values.length) {
-            ensureResultRows();
-        }
+        ensureValuesCapacity(position + 1, false);
         values[position] = doubleToLongBits(value);
         if (valueIsNull != null) {
             valueIsNull[position] = false;
         }
+        numResults++;
     }
 
-    void ensureResultRows()
+    @Override
+    protected void ensureValuesCapacity(int capacity, boolean includeNulls)
     {
-        if (values.length <= numValues + numResults) {
-            values = Arrays.copyOf(values, Math.max(numValues + numResults + 10, values.length * 2));
-            block = null;
+        if (values.length < capacity) {
+            values = Arrays.copyOf(values, Math.max(capacity + 10, values.length * 2));
+            if (valueIsNull != null) {
+                valueIsNull = Arrays.copyOf(valueIsNull, values.length);
+            }
         }
-        if (valueIsNull != null) {
-            valueIsNull = Arrays.copyOf(valueIsNull, values.length);
-            block = null;
+        if (includeNulls && valueIsNull == null) {
```

Comment:
Wouldn't we want to resize `valueIsNull` to match the size of `values.length` if the array is not null?